### DISCUSSION TITLE
getUrl / getFullUrl doesn't make sense with `public` disk

### DISCUSSION THREAD

**Discussion by volkv** [spatie/laravel-medialibrary#1773] [Ideas]
Hi! Since public disk is configured like

```php       
 'public' => [
            'driver'     => 'local',
            'root'       => storage_path('app/public'),
            'url'        => env('APP_URL').'/storage',
            'visibility' => 'public',
        ]
```
`Media->getUrl()` calls `Storage::disk('public')->url` which always returns a full URL (domain+path)

so it makes no sense having `getFullUrl` method and gives no out-of-the-box way to get relative URL

would be nice to have something like getRelativeUrl and getFullUrl doing its jobs

Thanks

**Comment by vesper8** [comment]
I second the need for a `getRelativeUrl`

I just found what I believe to be a bug. `getUrl()` is supposed to return the path to the image without the domain according to the documentation, as opposed to `getFullUrl`. In my case it does indeed work this way on both `local` and `production`, but when I serve my app through `ngrok` for some inexplicable reason only in this instance `getUrl` returns the full url including the domain ???

Having a `getRelativeUrl` would be clearer and hopefully would ensure that this odd behaviour never occurs

**Comment by freekmurze** [comment]
Feel free to PR improvements around this. Make sure to include tests and updated docs. 👍
